FirebaseFunctions/Sources/Functions.swift

```diff
@@ -165,14 +129,89 @@ internal enum FunctionsConstants {
    * @param host The host of the local emulator, such as "localhost".
    * @param port The port of the local emulator, for example 5005.
    */
-  @objc public func useEmulator(withHost host: String, port: Int) {
+  @objc open func useEmulator(withHost host: String, port: Int) {
     let prefix = host.hasPrefix("http") ? "" : "http://"
     let origin = String(format: "\(prefix)\(host):%li", port)
     emulatorOrigin = origin
   }
 
+  // MARK: - Public APIs only for Objective C
+
+  /**
+   * Creates a Cloud Functions client or returns a pre-existing instance if it already exists.
+   */
+  @objc(functions) open class func __functions() -> Functions {
+    return functions()
+  }
+
+  /**
+   * Creates a Cloud Functions client with the given app, or returns a pre-existing
+   * instance if one already exists.
+   * @param app The app for the Firebase project.
+   */
+  @objc(functionsForApp:) open class func __functionsForApp(app: FirebaseApp) -> Functions {
+    return functions(app: app)
+  }
+
+  /**
+   * Creates a Cloud Functions client with the default app and given region.
+   * @param region The region for the http trigger, such as "us-central1".
+   */
+  @objc(functionsForRegion:) open class func __functionsForRegion(region: String) -> Functions {
+    return functions(region: region)
+  }
+
+  /**
+   * Creates a Cloud Functions client with the given app and region, or returns a pre-existing
+   * instance if one already exists.
+   * @param customDomain A custom domain for the http trigger, such as "https://mydomain.com".
+   */
+  @objc(functionsForCustomDomain:) open class func __functionsForCustomDomain(customDomain: String?)
+    -> Functions {
+    return functions(customDomain: customDomain)
+  }
+
   // MARK: - Private Funcs (or Internal for tests)
 
+  @objc internal init(projectID: String,
+                      region: String,
+                      customDomain: String?,
+                      auth: AuthInterop?,
+                      messaging: MessagingInterop?,
+                      appCheck: AppCheckInterop?,
+                      fetcherService: GTMSessionFetcherService = GTMSessionFetcherService()) {
+    self.projectID = projectID
+    self.region = region
+    self.customDomain = customDomain
+    emulatorOrigin = nil
+    contextProvider = FunctionsContextProvider(auth: auth,
+                                               messaging: messaging,
+                                               appCheck: appCheck)
+    self.fetcherService = fetcherService
+  }
+
+  internal convenience init(app: FirebaseApp,
```

Comment:
Nit: can you add a small doc comment here mentioning this uses the component system for initialization?

Reply:
Done.

Reply:
Small nit for next time you're editing (no need to kick off tests again for this small change): Can you make it a doc comment with `///` instead of a regular one?